src/layers/core/hexagon-layer/hexagon-layer.js

```diff
@@ -54,19 +55,23 @@ const defaultProps = {
 };
 
 function _needsReProjectPoints(oldProps, props) {
-  return oldProps.radius !== props.radius;
+  return oldProps.radius !== props.radius || oldProps.hexagonAggregator !== props.hexagonAggregator;
 }
 
 function _percentileChanged(oldProps, props) {
   return oldProps.lowerPercentile !== props.lowerPercentile ||
     oldProps.upperPercentile !== props.upperPercentile;
 }
 
+function _needsReSortBins(oldProps, props) {
+  return oldProps.colorValueAccessor !== props.colorValueAccessor;
+}
+
 export default class HexagonLayer extends Layer {
   constructor(props) {
-    if (!props.radius) {
-      log.once(0, 'HexagonLayer: radius in meter is needed to aggregate points into ' +
-        'hexagonal bins, Now using 1000 meter as default');
+    if (!props.radius || !typeof props.hexagonAggregator === 'function') {
```

Comment:
> if user pass in a hexagonAggregator then radius is not required. If not, our default hexagonAggregator requires radius to be set.

OK that makes a lot of sense! Perhaps the condition and the error message could be rewritten to make this a little clearer? Use whatever makes most sense to you.
```
if (!hexagonAggregator && !radius) {
   log.once('Default hexagonAggregator requires radius prop to be set);
}
```